.devcontainer/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM mcr.microsoft.com/devcontainers/cpp:1-ubuntu-22.04
+
+USER vscode
+
+# Install latest cmake
+RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | sudo tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null
+RUN echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ jammy main' | sudo tee /etc/apt/sources.list.d/kitware.list >/dev/null
+RUN sudo apt-get update && sudo apt-get install -y cmake
+
+# Install pre-commit
+RUN sudo apt-get install -y python3-pip && pip3 install pre-commit
+
+# Avoid ASAN Stalling
+# Alternative is to update to clang-18 and gcc-13.2
+# Maybe crashing codespace???
```

Comment:
Can we convert to an actual statem/link with an issue?

Reply:
surething, should I try to install the latest stable clang and gcc here?

Reply:
That's quite a lot of compilers to install, my gut tells me there will be dependency issues if I don't install from source.

I am not sure if codespace caches the underlying devcontainer image, and from my own testing it is a little bit hit-or-miss, I worry if not a lot of people use codespace/ GitHub does not cache the underlying image for whatever reason. Library consumer will need to wait a long time for the devcontainer to boot up due to the size of the `Dockerfile`.

In fact I specifically didn't install the latest version of clang and gcc by choice.

I would suggest let's revisit this later and see how good codespace caching is. Let me know if I should have the latest clang and gcc installed though.

Reply:
done with the comment update.